src/serialization/interval_log/mod.rs

```diff
@@ -12,58 +12,102 @@
 //! parse) logs with intervals in any order, but the expectation is that they will be sorted.
 //!
-//! To parse a log, see `IntervalLogIterator`. To write a log, see `IntervalLogHeaderWriter`.
+//! To parse a log, see `IntervalLogIterator`. To write a log, see `IntervalLogWriterBuilder`.
 //!
 //! # Timestamps
 //!
 //! Each interval has a timestamp in seconds associated with it. However, it's not necessarily as
-//! simple as just interpreting the number as seconds since the epoch.
-//!
-//! There are two optional pieces of header metadata: "StartTime" and "BaseTime". Neither, one, or
-//! both of these may be present. It is possible to have multiple StartTime or BaseTime entries in
-//! the log, but that is discouraged as it is confusing to interpret. It is also possible to have
-//! StartTime and BaseTime interleaved with interval histograms, but that is even more confusing, so
-//! this API prevents you from doing so.
-//!
-//! ### Timestamp options
-//!
-//! This is a summary of the logic used by the Java impl's `HistogramLogReader` for StartTime and
-//! BaseTime.
-//!
-//! - Neither are present: interval timestamps should be interpreted as seconds since the epoch.
+//! simple as just interpreting the number as seconds since the epoch. There are two optional pieces
+//! of header metadata: "StartTime" and "BaseTime". Neither, one, or both of these may be present.
+//! It is possible to have multiple StartTime or BaseTime entries in the log, or even interleaved
+//! with interval histograms, but that is confusing, so this API prevents you from doing so.
+//!
+//! When BaseTime is present, per-interval timestamps are the number of seconds since BaseTime. When
+//! it is absent, the per-interval timestamps represent fractional seconds since the epoch (aka
+//! Unix time). BaseTime is useful because it allows you to have somewhat human-readable timestamps
+//! for each interval -- it's easier to see 245.3 and see that that's 4 minutes after the start of
+//! whatever it is you're doing than it is to mentally parse a Unix timestamp. Naturally, you can
+//! always calculate the deltas after the fact if you're willing to write a little tooling, but in
+//! some cases log files are consumed by humans too.
+//!
+//! While BaseTime is used to redefine per-interval timestamps for human readability, StartTime
+//! provides information about when the process that is generating the log reached some "start"
+//! condition. It's frequently the case that intervals will start some seconds after the process
+//! started, whether due to initial warmup before a benchmark or because it just takes a
+//! while to start up. If this is relevant to your workload, use StartTime to record the actual
+//! process start time (or other relevant "start" condition, like when a benchmark begins on an
+//! already long-running process). You could then use this when processing a log to more accurately
+//! plot interval data over time.
```

Comment:
Are the per-interval timestamps then relative to `StartTime`?

Reply:
From reading the example below, I take it they aren't (that is, they are *always* relative to `BaseTime` or UNIX epoch). We might want to clarify that here.